### ISSUE TITLE
Incorrect rotation or EXIF orientation of portrait photos captured with collect app

### ISSUE THREAD

**Issue by jbp35** [getodk/collect#4571]
#### Software and hardware versions 
ODK Collect v1.30.1
Android 9
Xiaomi Mi A1

#### Problem description

Photos captured in portrait format with ODK Collect are not saved or rendred in the right direction. 

Portrait format with ODK Collect:
![Portrait_ODK_Collect](https://user-images.githubusercontent.com/19955142/119104336-6e75da00-ba1c-11eb-97d9-0b132fe547f9.jpg)

Portrait format with camera app:
![Portrait_Camera app](https://user-images.githubusercontent.com/19955142/119104333-6ddd4380-ba1c-11eb-845c-25cb2e7b5baf.jpg)

Landscape format with ODK Collect:
![Landscape_ODK_Collect](https://user-images.githubusercontent.com/19955142/119104327-6d44ad00-ba1c-11eb-9ebf-39dcbf4f691c.jpg)

Landscape format with camera app:
![Landscape_Camera app](https://user-images.githubusercontent.com/19955142/119104307-67e76280-ba1c-11eb-98b4-b4145f99628f.jpg)

#### Steps to reproduce the problem

1. Create a simple form containing a photo uploader
2. Using ODK Collect, take a photo on a device that produces an EXIF Orientation tag for portrait photos and upload it via the form
3. Download the photo from KoBoToolbox or access the file directly in the folder : "Internal/Android/data/org.odk.collect.android/files/instances/<instance directory>/" 
4. Open on any system that understands EXIF Orientation and notice that landscape photos render correctly, but portrait photos are sideways
5. Compare with the same photo taken directly with the camera app of the device where portrait photos are displayed vertical as expected

#### Other information 

This issue was initially posted on : 
https://github.com/kobotoolbox/kobocat/issues/707

**Comment by lognaturel** [comment]
Thanks! A PR would be much appreciated.

**Comment by abhishekab** [comment]
@lognaturel I would love to make this as my first issue, can I take this up?

I have identified the problem here. It is basically an issue in utilities/ImageConvertor class. So there is code written to explicitly rotate the image if EXIF rotation is there, this was done to solve Issue [#36](https://github.com/getodk/collect/issues/36). But it actually introduced a problem. So the code rotates the image and transforms it to new rotated image, but at the same time does not remove EXIF rotation. So while in the ODK app, since the ImageView is not sensitive to EXIF data, the transformed image shows correctly. But for external image viewers which are sensitive to EXIF does rotation to the already transformed image and thus a double rotation is done on the original image, which disorients the image.

**Comment by abhishekab** [comment]
@getodk-bot claim

**Comment by lognaturel** [comment]
Thanks so much for that detailed analysis! We'd very much appreciate your help with this issue. 👍

**Comment by abhishekab** [comment]
Thanks a lot @lognaturel . Excited to work on this. I can think of two main approaches to solve this:
1) Remove the explicit bitmap transformation which rotates the image, and preserve EXIF orientation. And make the ImageView displaying the image in ImageWidget sensitive to EXIF just for presentation.

2) Let the explicit bitmap transformation as it is and let it rotate the image, and at the same time remove the orientation from EXIF before saving the transformed image.

I feel (1) is a more correct thing to do as why transform an image when it is not needed. Also I noticed in my few experiments I did that bitmap transformation done to rotate the image is actually increasing the size of the image by a significant factor, which I feel is undesirable.

Wanted to hear your thoughts as well.
